src/utils.jl

```diff
@@ -129,6 +129,23 @@ _scopename(sym) = sym
 _scopename(parent, child) = Expr(:., parent, QuoteNode(child))
 _scopename(parent, child, rest...) = Expr(:., parent, _scopename(child, rest...))
 
+function must_be_toplevel(mod::Module, ex::Expr; expand::Bool=true)
+    head = ex.head
+    if head === :toplevel || head === :block
+        for a in ex.args
+            if isa(a, Expr)
+                must_be_toplevel(mod, a) && return true
+            end
+        end
+    elseif head === :export || head === :import || head === :using || head === :global || head === :const
+        return true
```

Comment:
It feels like `must_be_toplevel` actually means `must_be_just_evaluated` or something ?
Because it seems like what should always be in toplevel is actually only `:const`, and the others can work within local scopes (like `let`), like:
```julia
julia> let 
           i = 0
           struct A
               val
               A(v = i) = new(v)
           end
           export A
       end

julia> A()
A(0)
```

Reply:
Yeah, in a way I guess it's really `doesnt_lower`.

Reply:
Thanks for pushing on this. It turns out that if you wrap in a `:block` together with a `LineNumberNode`, then even `export`, `import`, and `using` do lower to a `CodeInfo`. And `:const` already did, I'm not sure why I added it here.

That leaves only `:global` expressions. Does anyone know how to make them lower to a CodeInfo?

Given that it's just `:global`, I've reworked this to simply not deliver a `:block`/`:toplevel` wrapper when it's a `:global` expr.

Reply:
> And `:const` already did

I never know that, so sneaky.

> Given that it's just `:global`, I've reworked this to simply not deliver a `:block`/`:toplevel` wrapper when it's a `:global` expr.

Yeah, the docstring is kind enough, I think.